test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs

```diff
@@ -57,6 +60,18 @@ public static IEnumerable<object[]> ExpressionAndTexts
                         string.Empty
                     },
                     {
+                        (Expression<Func<TestModel, int>>)(model => models[0].SelectedCategory.CategoryId),
+                        "models[0].SelectedCategory.CategoryId"
+                    },
+                    {
+                        (Expression<Func<TestModel, string>>)(model => modelTest.Name),
+                        "modelTest.Name"
+                    },
+                    {
+                        (Expression<Func<TestModel, Type>>)(model => modelType),
+                        "modelType"
+                    },
+                    {
```

Comment:
We'd have to test those cases. But, this goes beyond the reported problem. So we should focus the follow-up discussion on `dev`.